File: Python DSA/Bitmanipulation.py
# Example: Counting the number of 1s in the binary representation of a number
def count_set_bits(n):
    count = 0
    while n:
        count += n & 1
        n >>= 1
    return count

n = 29  # Binary: 11101
print(count_set_bits(n))  # Output: 4

File: C++ DSA/Bitmanipulation.cpp
#include <iostream>
using namespace std;

int countSetBits(int n) {
    int count = 0;
    while (n > 0) {
        count += (n & 1);
        n >>= 1;
    }
    return count;
}

int main() {
    int n = 29; // Binary: 11101
    cout << "Number of set bits: " << countSetBits(n) << endl; // Output: 4
    return 0;
}
